pkg/chunk/storage/factory.go

```diff
@@ -163,7 +166,7 @@ func NewStore(cfg Config, storeCfg chunk.StoreConfig, schemaCfg chunk.SchemaConf
 		if objectStoreType == "" {
 			objectStoreType = s.IndexType
 		}
-		chunks, err := NewChunkClient(objectStoreType, cfg, schemaCfg)
+		chunks, err := NewChunkClient(objectStoreType, cfg, schemaCfg, reg)
```

Comment:
Here you should wrap the reg with a different label value, like `chunks-client-` + `s.From.String()`.